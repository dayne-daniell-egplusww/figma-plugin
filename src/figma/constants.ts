export const MENU_COMMANDS = {
  SHOW_UPLOAD_UI: 'show_upload_ui',
  SHOW_PROCESS_UI: 'show_process_ui',

  RESET: 'reset',
};

export const CLIENT_STORAGE_KEYS = {
  GITLAB_USER_CONFIG: 'GITLAB_ACCESS_TOKEN', // GITLAB_ACCESS_TOKEN must be maintained for legacy reasons
};
