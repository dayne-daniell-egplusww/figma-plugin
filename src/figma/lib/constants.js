export const DELIMITER = '___';

export const SLIDE_ID_FROM_NAME = '_';

export const SPLIT_DELIMITER = '_';

export const TAB_OVERLAY_ID_FROM_NAME = '__';

export const SLIDE_KEY = 'Slide_';

export const TAB_KEY = 'tab-';

export const OVERLAY_KEY = 'overlay-';

export const SLIDE_CONTENT = 'CONTENT';

export const SLIDE_CHROME = 'CHROME';

export const SLIDE_TEMPLATE = 'TEMPLATE';
    
export const TRIGGERS = 'ON_CLICK'; // space delimited

export const TRIGGER_DELIMITER = '---'; // space delimiter

export const PRES_SLIDE_DELIMITER = '--'; // space delimiter

export const PRES_SLIDE_NO_ID = 'PRES_SLIDE_NO_ID'; // space delimiter