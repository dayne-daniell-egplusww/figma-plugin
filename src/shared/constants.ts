export const FIGMA_MESSAGE_TYPES = {
  UPDATE_SELECTION: 'UPDATE_SELECTION',
  SET_USER_CONFIG: 'SET_USER_CONFIG',
  EXPORT_SELECTION: 'EXPORT_SELECTION',
  PROCESS_SELECTION: 'PROCESS_SELECTION',
  PACKAGE_SELECTION: 'PACKAGE_SELECTION',
  RESIZE: 'RESIZE',
  NOTIFY: 'NOTIFY',
  WARNING_MESSAGE: 'WARNING_MESSAGE',
  ERROR_MESSAGE: 'ERROR_MESSAGE',
  STOP_PLUGIN: 'STOP_PLUGIN'
};

export const DEFAULT_WINDOW_SIZE = {
  width: 380,
  height: 560,
};

export const GITLAB_INSTANCE_PROTOCOL = process.env.GITLAB_INSTANCE_PROTOCOL || 'https:';
