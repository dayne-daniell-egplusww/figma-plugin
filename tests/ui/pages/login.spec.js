import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueApollo from 'vue-apollo';
import { createMockClient } from 'mock-apollo-client';
import Login from '~/ui/pages/login.vue';
import FigmaAlert from '~/ui/components/figma_plugin_ds/alert.vue';
import currentUserQuery from '~/ui/graphql/queries/current_user.query.graphql';
import typeDefs from '~/ui/graphql/typedefs.graphql';
import { FIGMA_MESSAGE_TYPES } from '~/shared/constants';

const currentUserMock = {
  name: 'Waylon Smithers',
  username: 'wsmithers',
  id: 123,
};

const currentUserMockResponse = {
  data: {
    currentUser: {
      __typename: 'User',
      ...currentUserMock,
    },
  },
};

const localVue = createLocalVue();
localVue.use(VueApollo);

describe('Login page', () => {
  let wrapper;
  let mockClient;
  let apolloProvider;
  let requestHandlers;
  let $router;

  function createComponent({ data = {} } = {}, apolloHandlers) {
    requestHandlers = {
      currentUserQuery: jest.fn().mockResolvedValue(currentUserMockResponse),
      ...apolloHandlers,
    };

    mockClient = createMockClient({ typeDefs });
    mockClient.setRequestHandler(currentUserQuery, requestHandlers.currentUserQuery);

    apolloProvider = new VueApollo({
      defaultClient: mockClient,
    });

    $router = {
      push: jest.fn(),
    };

    wrapper = shallowMount(Login, {
      localVue,
      apolloProvider,
      data() {
        return {
          ...data,
        };
      },
      stubs: {
        'empty-issue-illustration': true,
        'gitlab-logo': true,
      },
      mocks: {
        $router,
      },
    });
  }

  describe('on initial load', () => {
    beforeEach(() => {
      createComponent();
    });

    it('renders splash screen', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('after initial load', () => {
    const mockInitialLoadState = { afterInitialLoad: true };

    it('renders login page', async () => {
      createComponent({ data: { ...mockInitialLoadState } });
      await wrapper.vm.$nextTick();

      expect(wrapper.html()).toMatchSnapshot();
    });

    describe('when access token is invalid', () => {
      beforeEach(async () => {
        createComponent(
          { data: { ...mockInitialLoadState } },
          {
            currentUserQuery: jest.fn().mockRejectedValue(),
          },
        );

        await wrapper.vm.$nextTick();
      });

      it('renders error message', () => {
        expect(wrapper.findComponent(FigmaAlert).exists()).toBe(true);
      });
    });

    describe('when access token is valid', () => {
      const postMessageSpy = jest.fn();

      beforeEach(async () => {
        window.parent.postMessage = postMessageSpy;

        createComponent({
          data: {
            ...mockInitialLoadState,
            userConfig: { accessToken: '123abc', gitlabInstance: { host: 'mygitlab.com' } },
          },
        });
        await wrapper.vm.$nextTick();
      });

      afterEach(() => {
        postMessageSpy.mockClear();
      });

      it('navigates to UploadSelection page', () => {
        expect($router.push).toHaveBeenCalledTimes(1);
        expect($router.push).toHaveBeenCalledWith({ name: 'UploadSelection' });
      });

      it('requests Figma to store access token', () => {
        expect(postMessageSpy).toHaveBeenCalledTimes(1);
        expect(postMessageSpy).toHaveBeenCalledWith(
          {
            pluginMessage: {
              data: { accessToken: '123abc', gitlabInstance: { host: 'mygitlab.com' } },
              type: FIGMA_MESSAGE_TYPES.SET_USER_CONFIG,
            },
          },
          '*',
        );
      });
    });
  });
});
