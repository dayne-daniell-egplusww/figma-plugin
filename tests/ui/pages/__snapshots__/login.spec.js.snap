// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page after initial load renders login page 1`] = `
"<main class=\\"h-100 d-flex flex-column\\">
  <figma-section-stub class=\\"flex-grow\\">
    <!---->
    <div class=\\"w-100 p-medium m-auto\\">
      <empty-issue-illustration-stub></empty-issue-illustration-stub>
    </div>
    <p class=\\"text-xsmall\\">You are one step away from contributing to your designs.</p>
    <p class=\\"text-xsmall\\">
      Authenticate with GitLab using a GitLab personal access token.
    </p>
    <form id=\\"set_user_config_form\\" class=\\"m-0\\">
      <figma-label-stub>GitLab access token</figma-label-stub>
      <figma-input-stub value=\\"\\" state=\\"true\\" id=\\"access_token_input\\" type=\\"password\\" required=\\"\\"></figma-input-stub>
      <figma-checkbox-stub label=\\"I am using a self-managed GitLab instance\\" id=\\"gitlab_instance_toggle\\" class=\\"pt-small pb-small\\"></figma-checkbox-stub>
      <!---->
    </form>
  </figma-section-stub>
  <figma-section-stub class=\\"d-flex justify-content-between align-items-center w-100 border-top-grey\\"><a href=\\"https://gitlab.com/gitlab-org/gitlab-figma-plugin/-/wikis/home#creating-a-gitlab-personal-access-token\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"text-xsmall\\">Where to find my token?
    </a>
    <figma-button-stub variant=\\"primary\\" type=\\"submit\\" form=\\"set_user_config_form\\">Save settings</figma-button-stub>
  </figma-section-stub>
</main>"
`;

exports[`Login page on initial load renders splash screen 1`] = `
"<main class=\\"h-100 d-flex flex-column\\">
  <div class=\\"h-100 d-flex align-items-center justify-content-center\\">
    <gitlab-logo-stub class=\\"w-50\\"></gitlab-logo-stub>
  </div>
</main>"
`;
